lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -438,6 +439,7 @@ ExhaustiveSynthesis::synthesize(SMTLIBSolver *SMTSolver,
   std::vector<Inst *> Vars;
   findVars(LHS, Vars);
 
+  unsigned ValidRHS = 0;
```

Comment:
why do you want a separate variable for this? doesn't it work to inspect the length of the vector?